### PR TITLE
Adds pebble mutating webhook

### PR DESCRIPTION
**Pull request by claudiubelu**
This is required for the cases in which the containers have a read-only root filesystem, which means that Pebble couldn't create its necessary files otherwise.

The webhook only affects containers with `readOnlyRootFilesystem=true` security context. The webhook mutates the Pod request to add an empty dir volume for Pebble to use to store its data for the mentioned containers. Additionally, it adds / amends the `$PEBBLE` and `$PEBBLE_COPY_ONCE` environment variables accordingly for the affected containers. If there are no such containers, the webhook will essentially be noop.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+*/
+
+package main
+
+import (
+	"errors"
+	"net/http"
+	"os"
+
+	log "github.com/sirupsen/logrus"
```
Comment (bschimke95) [inline cmd/main.go]: nit: please use the standard log library. `logrus` is in maintenance mode. https://github.com/sirupsen/logrus

I know that we have some left-over usage in our codebase but for new projects we should not use it anymore.

File: pkg/webhook/mutation.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+		return nil, fmt.Errorf("Request could not be decoded: %v", err)
+	}
+
+	return &pod, nil
+}
+
+// Check if the given container has the given mount path.
+// Add an empty dir volume for Pebble to store its state in.
+func addPebbleMountMutation(ar *v1.AdmissionReview) (*v1.AdmissionResponse, error) {
+	log.Debug("calling addPebbleMountMutation")
```
Comment (bschimke95) [inline pkg/webhook/mutation.go]: nit: left-over debugging output?

File: pkg/webhook/mutation.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+	podResource := metav1.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+	if ar.Request.Resource != podResource {
+		err := fmt.Errorf("Expected resource to be: '%s', actual: '%s'", podResource, ar.Request.Resource)
+		return nil, err
```
Comment (bschimke95) [inline pkg/webhook/mutation.go]: just directly return the error and please return errors in lowercase https://staticcheck.dev/docs/checks#ST1005

File: pkg/webhook/mutation.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+	}
+
+	pod, err := getPod(ar)
+	if err != nil {
+		return nil, err
```
Comment (bschimke95) [inline pkg/webhook/mutation.go]: please wrap errors: `fmt.Errorf("failed to get pod: %w", err)`

(here and elsewhere)

**Review by bschimke95 — COMMENTED** [review]
Looks good, minor comments

File: cmd/main.go (hunk 1 above)
Comment (claudiubelu) [reply cmd/main.go]: Done.

File: pkg/webhook/mutation.go (hunk 2 above)
Comment (claudiubelu) [reply pkg/webhook/mutation.go]: Not really, but I can remove it.

File: pkg/webhook/mutation.go (hunk 3 above)
Comment (claudiubelu) [reply pkg/webhook/mutation.go]: Done

File: pkg/webhook/mutation.go (hunk 4 above)
Comment (claudiubelu) [reply pkg/webhook/mutation.go]: Done

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**